// Copyright 2023 Google LLC
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//     http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.

syntax = "proto3";

package distributed_point_functions;

import "dpf/distributed_point_function.proto";

// For faster allocations of sub-messages.
option cc_enable_arenas = true;

message SeedCorrectionOtSenderMessage {
  repeated Block masked_message_one = 1;
  repeated Block masked_message_two = 2;
}

message SeedCorrectionOtReceiverMessage {
  repeated bool choice_bit_mask = 1;
}

message SeedCorrectionShare {
  repeated Block seed = 1;
  repeated bool control_bit_left = 2;
  repeated bool control_bit_right = 3;
}

message ValueCorrectionOtSenderMessage {
  repeated Value masked_message_one = 1;
  repeated Value masked_message_two = 2;
}

message ValueCorrectionOtReceiverMessage {
  repeated bool choice_bit_mask = 1;
}

message ValueCorrectionShare {
  repeated Value value = 1;
}
